GitVersionCore/BuildServers/AppVeyor.cs

```diff
@@ -18,14 +18,17 @@ public override bool CanApplyToCurrentContext()
             return !string.IsNullOrEmpty(Environment.GetEnvironmentVariable("APPVEYOR"));
         }
 
-        public override void PerformPreProcessingSteps(string gitDirectory)
+        public override void PerformPreProcessingSteps(string gitDirectory, bool noFetch)
         {
             if (string.IsNullOrEmpty(gitDirectory))
             {
                 throw new WarningException("Failed to find .git directory on agent.");
             }
 
-            GitHelper.NormalizeGitDirectory(gitDirectory, authentication);
+            if (!noFetch)
```

Comment:
To save this duplication can we pass it down to NormalizeGitDirectory, otherwise it will be easy to forget this step with new Build Servers